### PR TITLE
drivers: wifi: Get extended capabilities

### PR DESCRIPTION
**Pull request by sr1dh48r**
[SHEL-1269]: Multiple BSSID bit is not set in Association request
- Implement get_ext_capab op to get MBSSID support cap. info
- Update max_num_rx_agg_sessions to 8

Signed-off-by: Sridhar Nuvusetty <sridhar.nuvusetty@nordicsemi.no>

### CODE REVIEW

File: drivers/wifi/nrf700x/zephyr/src/zephyr_fmac_main.c [hunk 1]
```diff
@@ -53,7 +53,7 @@ BUILD_ASSERT(RPU_PKTRAM_SIZE >=
 static const unsigned char aggregation = 1;
 static const unsigned char wmm = 1;
 static const unsigned char max_num_tx_agg_sessions = 4;
-static const unsigned char max_num_rx_agg_sessions = 2;
+static const unsigned char max_num_rx_agg_sessions = 8;
```
Comment (krish2718) [inline drivers/wifi/nrf700x/zephyr/src/zephyr_fmac_main.c:56]: is this related?

File: drivers/wifi/nrf700x/zephyr/src/zephyr_wpa_supp_if.c [hunk 2]
```diff
@@ -1247,6 +1247,34 @@ enum wifi_nrf_status wifi_nrf_parse_sband(
...
 }
 
+static void wiphy_info_extended_capab_cfg(struct wifi_nrf_ext_capa *capa,
+					  struct nrf_wifi_event_get_wiphy *wiphy_info)
+{
+	unsigned int len;
+
+	capa->iftype = NRF_WIFI_IFTYPE_STATION;
+
+	wpa_printf(MSG_DEBUG,
```
Comment (krish2718) [inline drivers/wifi/nrf700x/zephyr/src/zephyr_wpa_supp_if.c:1257]: these should be replaced with LOG_ macros

File: drivers/wifi/nrf700x/zephyr/src/zephyr_wpa_supp_if.c [hunk 3]
```diff
@@ -1273,6 +1301,34 @@ void wifi_nrf_wpa_supp_event_get_wiphy(void *if_priv,
 							   &band);
 	}
 
+	if ((wiphy_info->params_valid & NRF_WIFI_GET_WIPHY_VALID_EXTENDED_CAPABILITIES) &&
+	    rpu_ctx_zep->extended_capa == NULL) {
+		rpu_ctx_zep->extended_capa = os_malloc(wiphy_info->extended_capabilities_len);
+		if (rpu_ctx_zep->extended_capa) {
+			os_memcpy(rpu_ctx_zep->extended_capa, wiphy_info->extended_capabilities,
```
Comment (krish2718) [inline drivers/wifi/nrf700x/zephyr/src/zephyr_wpa_supp_if.c:1308]: these should also be replaced with direct Zephyr calls.

**Review by krish2718 — CHANGES_REQUESTED** [review]

File: drivers/wifi/nrf700x/zephyr/inc/zephyr_fmac_main.h [hunk 4]
```diff
@@ -78,6 +78,13 @@ struct wifi_nrf_ctx_zep {
 	struct rpu_conf_params conf_params;
 #endif /* CONFIG_NRF700X_WIFI_UTIL */
 #endif /* CONFIG_NRF700X_RADIO_TEST */
+	u8 *extended_capa, *extended_capa_mask;
```
Comment (sachinthegreen) [inline drivers/wifi/nrf700x/zephyr/inc/zephyr_fmac_main.h:81]: Use `unsigned char` instead of `u8` to be consistent with all other elements of similar type in the structure.

File: drivers/wifi/nrf700x/zephyr/inc/zephyr_fmac_main.h [hunk 5]
```diff
@@ -78,6 +78,13 @@ struct wifi_nrf_ctx_zep {
...
+	unsigned int extended_capa_len;
+	struct wifi_nrf_ext_capa {
+		enum nrf_wifi_iftype iftype;
+		u8 *ext_capa, *ext_capa_mask;
+		unsigned int ext_capa_len;
+	} iface_ext_capa[NRF_WIFI_IFTYPE_MAX];
```
Comment (sachinthegreen) [inline drivers/wifi/nrf700x/zephyr/inc/zephyr_fmac_main.h:87]: Why not just move this to `struct wifi_nrf_vif_ctx_zep` ?

File: drivers/wifi/nrf700x/zephyr/src/zephyr_wpa_supp_if.c [hunk 6]
```diff
@@ -1273,6 +1301,34 @@ void wifi_nrf_wpa_supp_event_get_wiphy(void *if_priv,
...
+				  wiphy_info->extended_capabilities_len);
+			wpa_hexdump(MSG_DEBUG,
+				    "nl80211: Driver-advertised extended capabilities(default)",
+				    rpu_ctx_zep->extended_capa, rpu_ctx_zep->extended_capa_len);
+		}
+		rpu_ctx_zep->extended_capa_mask = os_malloc(wiphy_info->extended_capabilities_len);
```
Comment (sachinthegreen) [inline drivers/wifi/nrf700x/zephyr/src/zephyr_wpa_supp_if.c:1314]: Please insert blank lines between statements to make code more readable.

File: drivers/wifi/nrf700x/zephyr/src/zephyr_wpa_supp_if.c [hunk 7]
```diff
@@ -1247,6 +1247,34 @@ enum wifi_nrf_status wifi_nrf_parse_sband(
 	return WLAN_STATUS_SUCCESS;
 }
 
+static void wiphy_info_extended_capab_cfg(struct wifi_nrf_ext_capa *capa,
```
Comment (sachinthegreen) [inline drivers/wifi/nrf700x/zephyr/src/zephyr_wpa_supp_if.c:1250]: Please prepend function name with wifi_nrf.

**Review by sachinthegreen — CHANGES_REQUESTED** [review]

File: drivers/wifi/nrf700x/zephyr/src/zephyr_fmac_main.c (hunk 1 above)
Comment (sr1dh48r) [reply drivers/wifi/nrf700x/zephyr/src/zephyr_fmac_main.c:56]: Yeah, apparently required for the certification test. Details in https://projecttools.nordicsemi.no/jira/browse/SHEL-1269

File: drivers/wifi/nrf700x/zephyr/inc/zephyr_fmac_main.h (hunk 5 above)
Comment (sr1dh48r) [reply drivers/wifi/nrf700x/zephyr/inc/zephyr_fmac_main.h:87]: Since this is device/radio level capabilities.

Comment (sachinthegreen) [reply drivers/wifi/nrf700x/zephyr/inc/zephyr_fmac_main.h:93]: Ok...but these are still per interface so it would make sense to have them in the interface specific structure.

**Review by sachinthegreen — CHANGES_REQUESTED** [review]

File: drivers/wifi/nrf700x/zephyr/inc/zephyr_fmac_main.h (hunk 5 above)
Comment (sr1dh48r) [reply drivers/wifi/nrf700x/zephyr/inc/zephyr_fmac_main.h:93]: Ah ok you are referring to iface_ext_capa specifically. I only remember the radio specific params. Doesn't make much of a difference but ok, we can move those ..

**Review by sachinthegreen — APPROVED** [review]

**Review by krish2718 — APPROVED** [review]
